tests/test_repack.py

```diff
@@ -1399,15 +1397,6 @@ def test_when_table_has_negative_pk_values(
         )
 
 
-@pytest.mark.xfail(
-    raises=NumericValueOutOfRange,
-    reason="""
-E           psycopg.errors.NumericValueOutOfRange: integer out of range
-E           CONTEXT:  SQL function "psycopack_repacked_6723301_fun" statement 1
-E           SQL statement "SELECT "public"."psycopack_repacked_6723301_fun"(NEW."id", NEW."id")"
-E           PL/pgSQL function psycopack_repacked_6723301_tgr() line 4 at PERFORM
-""",
-)
```

Comment:
I'm wondering what happens here for other different types of primary keys and if the code handles them. I.e., if we parametrise the test like this:

```python
@pytest.mark.parametrize(
    "initial_pk_type",
    (
        "integer",
        "serial",
        "smallint",
        "smallserial",
    ),
)
```

Thoughts?

Reply:
I added another commit to cover the above. With slight adjustment to a test factory, the code handles all these PK types.